src/dahua.ts

```diff
@@ -28,51 +27,57 @@ class DahuaEvents {
     public RECONNECTING_EVENT_NAME: string = 'reconnecting'
 
     private host:                   string
+    private user:                   string
+    private pass:                   string
+    private requestConfig:          AxiosRequestConfig
 
     constructor(host: string, user: string, pass: string, useHttp: boolean) {
         this.host = host
-        const auth: AxiosBasicCredentials = {
-            username: user,
-            password: pass
-        }
+        this.user = user
+        this.pass = pass
 
-        let keepAliveAgent
-        if(useHttp) {
-            keepAliveAgent = new HttpAgent({
+        const keepAliveAgent: HttpAgent | HttpsAgent = useHttp ?
+            new HttpAgent({
                 keepAlive: this.AGENT_SETTINGS.keepAlive,
                 keepAliveMsecs: this.AGENT_SETTINGS.keepAliveMsecs,
                 maxSockets: this.AGENT_SETTINGS.maxSockets,
                 maxFreeSockets: this.AGENT_SETTINGS.maxFreeSockets,
-            }) 
-        } else {
-            keepAliveAgent = new HttpsAgent({
+            })
+            :
+            new HttpsAgent({
                 keepAlive: this.AGENT_SETTINGS.keepAlive,
                 keepAliveMsecs: this.AGENT_SETTINGS.keepAliveMsecs,
                 maxSockets: this.AGENT_SETTINGS.maxSockets,
                 maxFreeSockets: this.AGENT_SETTINGS.maxFreeSockets,
                 rejectUnauthorized: false,
                 minVersion: "TLSv1"
             })

```

Comment:
No worries will revert to if statement.

Although I just want to point out one big benefit of a ternary is that it's possible to use a `const` rather than a `let`, preventing accidental overwriting the variable somewhere else.